<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>WidgetLaserSetting</class>
 <widget class="QWidget" name="WidgetLaserSetting">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>319</width>
    <height>645</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="topMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Item</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Laser</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Type</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="customComboBox" name="comboBox_model">
        <property name="currentIndex">
         <number>-1</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_12">
     <property name="title">
      <string>Setting</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_5">
      <item>
       <layout class="QFormLayout" name="formLayout_2">
        <item row="0" column="0">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>positive</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="customCheckBox" name="checkBox_upside"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_useForLocalization">
          <property name="text">
           <string>forLocation</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="customCheckBox" name="checkBox_useForLocalization"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_3">
          <property name="toolTip">
           <string>Angle interval of laser beam</string>
          </property>
          <property name="text">
           <string>step[°]</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="customDoubleSpinBox" name="doubleSpinBox_step">
            <property name="toolTip">
             <string>bigger than 0</string>
            </property>
            <property name="decimals">
             <number>5</number>
            </property>
            <property name="minimum">
             <double>0.001000000000000</double>
            </property>
            <property name="maximum">
             <double>99999999.998999997973442</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
            <property name="value">
             <double>0.500000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_7">
          <property name="toolTip">
           <string>end scanning angle</string>
          </property>
          <property name="text">
           <string>endAngle[°]</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="customDoubleSpinBox" name="doubleSpinBox_max_angle">
            <property name="toolTip">
             <string>range[-180,180]</string>
            </property>
            <property name="minimum">
             <double>-180.000000000000000</double>
            </property>
            <property name="maximum">
             <double>180.000000000000000</double>
            </property>
            <property name="value">
             <double>90.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_6">
          <property name="toolTip">
           <string>start scanning angle</string>
          </property>
          <property name="text">
           <string>startAngle[°]</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="customDoubleSpinBox" name="doubleSpinBox_min_angle">
            <property name="toolTip">
             <string>range[-180,180]</string>
            </property>
            <property name="minimum">
             <double>-180.000000000000000</double>
            </property>
            <property name="maximum">
             <double>180.000000000000000</double>
            </property>
            <property name="value">
             <double>-90.000000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_10">
     <property name="title">
      <string>NetWork</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_10">
      <item>
       <layout class="QFormLayout" name="formLayout_3">
        <item row="0" column="0">
         <widget class="QLabel" name="label_IP">
          <property name="text">
           <string>IP</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_port">
          <property name="text">
           <string>Port</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="customSpinBox" name="spinBox_port">
          <property name="maximum">
           <number>65535</number>
          </property>
          <property name="value">
           <number>2111</number>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QIpAddressEdit" name="widget_IP" native="true"/>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Location</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="label_x">
          <property name="text">
           <string>x:[m]</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="customDoubleSpinBox" name="doubleSpinBox_x">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="minimum">
           <double>-10.000000000000000</double>
          </property>
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
          <property name="value">
           <double>0.200000000000000</double>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_y">
          <property name="text">
           <string>y:[m]</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="customDoubleSpinBox" name="doubleSpinBox_y">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="minimum">
           <double>-10.000000000000000</double>
          </property>
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_z">
          <property name="text">
           <string>z:[m]</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="customDoubleSpinBox" name="doubleSpinBox_z">
          <property name="frame">
           <bool>true</bool>
          </property>
          <property name="accelerated">
           <bool>false</bool>
          </property>
          <property name="keyboardTracking">
           <bool>true</bool>
          </property>
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="minimum">
           <double>-10.000000000000000</double>
          </property>
          <property name="maximum">
           <double>10.000000000000000</double>
          </property>
          <property name="singleStep">
           <double>0.010000000000000</double>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_r">
          <property name="text">
           <string>r:[°]</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="customDoubleSpinBox" name="doubleSpinBox_r">
          <property name="toolTip">
           <string>range[-180,180]</string>
          </property>
          <property name="decimals">
           <number>2</number>
          </property>
          <property name="minimum">
           <double>-180.000000000000000</double>
          </property>
          <property name="maximum">
           <double>180.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>customDoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>ModelCustomWidget.h</header>
  </customwidget>
  <customwidget>
   <class>customSpinBox</class>
   <extends>QSpinBox</extends>
   <header>ModelCustomWidget.h</header>
  </customwidget>
  <customwidget>
   <class>customComboBox</class>
   <extends>QComboBox</extends>
   <header>ModelCustomWidget.h</header>
  </customwidget>
  <customwidget>
   <class>customCheckBox</class>
   <extends>QCheckBox</extends>
   <header>ModelCustomWidget.h</header>
  </customwidget>
  <customwidget>
   <class>QIpAddressEdit</class>
   <extends>QWidget</extends>
   <header>QIpAddressEdit.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
